src/main/java/org/apache/maven/shared/dependency/analyzer/ClassFileVisitorUtils.java

```diff
@@ -77,45 +77,39 @@ private static void acceptJar(URL url, ClassFileVisitor visitor) throws IOExcept
                 String name = entry.getName();
                 // ignore files like package-info.class and module-info.class
                 if (name.endsWith(".class") && name.indexOf('-') == -1) {
-                    visitClass(name, in, visitor);
+                    // Jars(ZIP) always use / as the separator character
+                    visitClass(name, in, visitor, '/');
                 }
             }
         }
     }
 
     private static void acceptDirectory(File directory, ClassFileVisitor visitor) throws IOException {
-        if (!directory.isDirectory()) {
-            throw new IllegalArgumentException("File is not a directory");
-        }
-
-        DirectoryScanner scanner = new DirectoryScanner();
-
-        scanner.setBasedir(directory);
-        scanner.setIncludes(new String[] {"**/*.class"});
-
-        scanner.scan();
 
-        String[] paths = scanner.getIncludedFiles();
-
-        for (String path : paths) {
-            path = path.replace(File.separatorChar, '/');
-
-            File file = new File(directory, path);
+        List<Path> classFiles;
+        try (Stream<Path> walk = Files.walk(directory.toPath())) {
+            classFiles = walk.filter(path -> path.getFileName().toString().endsWith(".class"))
```

Comment:
There is a native `endsWith()`

Reply:
I'll have to check but I don't think that will work. The method you reference checks if a Path ends with another Path. e.g. that /usr/local/tmp/foo.class ends with a path for foo.class.  However, I don't think it would work for /usr/local/tmp/foo.class ending with the extension .class.

See https://stackoverflow.com/questions/20531247/how-to-check-the-extension-of-a-java-7-path

Reply:
Yes, indeed. It is just a wrapper.